lib/cli.js

```diff
@@ -99,6 +99,10 @@ exports.run = () => {
 		throw new Error(colors.error(figures.cross) + ' The TAP reporter is not available when using watch mode.');
 	}
 
+	if ((hasFlag('--watch') || hasFlag('-w')) && isCi) {
+		throw new Error(colors.error(figures.cross) + ' AVA will not run with the --watch (-w) option in CI, because CI processes should terminate, and with the --watch option, AVA will never terminate.');
```

Comment:
I agree it could be more succinct, but I like how it includes the reason for why it's not available. Users don't really like being told something and not knowing "why".